### ISSUE TITLE
`E3691` false positive when specify `EngineVersion` without minor

### ISSUE THREAD

**Issue by sir-sigurd** [aws-cloudformation/cfn-lint#3849]
### CloudFormation Lint Version

1.20.1

### What operating system are you using?

Mac

### Describe the bug

I get 
```
E3691 14 is not one of ['11.22', '11.22-rds.20240418', '11.22-rds.20240509', '11.22-rds.20240808', '12.15', '12.16', '12.17', '12.18', '12.19', '12.20', '12.21', '12.22', '13.11', '13.12', '13.13', '13.14', '13.15', '13.16', '13.17', '13.18', '14.10', '14.11', '14.12', '14.13', '14.14', '14.15', '14.9', '15.10', '15.4', '15.5', '15.6', '15.7', '15.8', '15.9', '16.1', '16.2', '16.3', '16.4', '16.5', '16.6', '17.1', '17.2']
delme-rds-test.yaml:8:7
```
on the reproduction template

### Expected behavior

Not get `E3691` because the reproduction template actually works beause RDS API allows to specify only major version, see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.DBVersions.html:

> If a major version is specified but a minor version is not, Amazon RDS defaults to a recent release of the major version you have specified. 

### Reproduction template

```yaml
Resources:
  MyDB:
    Type: 'AWS::RDS::DBInstance'
    Properties:
      DBInstanceClass: db.t3.micro
      AllocatedStorage: 20
      Engine: postgres
      EngineVersion: 14
      MasterUsername: test
      MasterUserPassword: testtest
```
